src/System.Private.CoreLib/shared/System/Collections/Generic/EqualityComparer.cs

```diff
@@ -16,14 +10,9 @@
 namespace System.Collections.Generic
 {
     [Serializable]
-    [TypeDependencyAttribute("System.Collections.Generic.ObjectEqualityComparer`1")]
     [System.Runtime.CompilerServices.TypeForwardedFrom("mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089")] 
-    public abstract class EqualityComparer<T> : IEqualityComparer, IEqualityComparer<T>
+    public abstract partial class EqualityComparer<T> : IEqualityComparer, IEqualityComparer<T>
```

Comment:
`IndexOf/LastIndexOf` methods are one of those low-value code-bloating optimizations that are ok for JIT, but that are poor trade-off for AOT. I expect that we would take a non-trivial binary bloat hit in CoreRT/ProjectN if we add them to the shared copy.

CoreRT/ProjectN has this factored differently in AOT-friendly way. It would be nice to port that factoring to CoreCLR before moving it to shared partition.

Reply:
ok, moved these optimizations to CoreCLR only